lib/split/user.rb

```diff
@@ -38,13 +38,13 @@ def max_experiments_reached?(experiment_key)
     end
 
     def cleanup_old_versions!(experiment)
-      keys = user.keys.select { |k| k.match(Regexp.new(experiment.name)) }
+      keys = user.keys.select { |k| k.match(Regexp.new("^#{experiment.name}(:|$)")) }
```

Comment:
Don't include cases like `exp805_v2` as "old versions" for `exp805`. We only want related keys like: `exp805` or `exp805:1` or `exp805:finished`.